src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -0,0 +1,218 @@
+import PushNotification from 'react-native-push-notification';
+import PushNotificationIOS from '@react-native-community/push-notification-ios';
+import messaging from '@react-native-firebase/messaging';
+import React, {useState} from 'react';
+import {Platform} from 'react-native';
+import {useNavigation} from '@react-navigation/core';
+import {useRecoilState} from 'recoil';
+
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import profileAtom from '../../../atom/profileAtom';
+import useChatUtilsHook from '../chat/useChatUtilsHook';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import useUserAuthHook from '../auth/useUserAuthHook';
+import TokenStorage, {ITokenEnum} from '../../../utils/storage/custom/tokenStorage';
+import {getAnonymousUserId, getUserId} from '../../../utils/users';
+
+const usePushNotificationHook = () => {
+  const isIos = Platform.OS === 'ios';
+
+  const navigation = useNavigation();
+  const {signedProfileId, anonProfileId} = useUserAuthHook();
+  const {localDb} = useLocalDatabaseHook();
+  const {fetchChannelDetail, setSelectedChannel} = useChatUtilsHook();
+  const [, setProfileAtom] = useRecoilState(profileAtom);
+
+  const [isLoadingFetchingChannelDetail, setIsLoadingFetchingChannelDetail] = useState(false);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The state variable `isLoadingFetchingChannelDetail` seems to be used only within this hook and does not seem to be returned for use elsewhere. If it's not needed outside, consider removing it.



<!-- This is an auto-generated comment by OSS CodeRabbit -->